include/lsst/afw/geom/SkyWcs.h

```diff
@@ -0,0 +1,205 @@
+    Other constructors may have to work harder.
+- For the output ast::SkyFrame set SkyRef=CRVAL (in radians) and SkyRefIs="Ignored".
+    This makes CRVAL easy to look up, and "Ignored" prevents CRVAL from being used as an offset.
+*/
+class SkyWcs : public Transform<Point2Endpoint, SpherePointEndpoint> {
+public:
+    SkyWcs(SkyWcs const &) = delete;
+    /// Move constructor
+    SkyWcs(SkyWcs &&) = default;
+    SkyWcs &operator=(SkyWcs const &) = delete;
+    /// Move assignment operator
+    SkyWcs &operator=(SkyWcs &&) = default;
+
+    /**
+    Construct a pure tangent SkyWcs
+
+    @param[in] crpix  Center of projection on the CCD using the LSST convention:
+                        0, 0 is the lower left pixel of the parent image
+    @param[in] crval  Center of projection on the sky
+    @param[in] cdMatrix  CD matrix, where element (i-1, j-1) corresponds to FITS keyword CDi_j
+                        and i, j have range [1, 2]
+    */
+    explicit SkyWcs(Point2D const &crpix, SpherePoint const &crval, Eigen::Matrix2d const &cdMatrix);
+
+    /**
+    Construct a WCS from FITS keywords
+
+    In addition to standard FITS WCS keywords uses these LSST-specific keywords:
+    LTV1, LTV2: offset of subimage with respect to parent image
+
+    TODO implement this using readFrameSet and adding full standardization
+
+    Currently does not strip the used keywords, but that capability should probably be added
+    */
+    explicit SkyWcs(daf::base::PropertyList &metadata);
+
+    /**
+    Construct a WCS from an ast::FrameSet
+
+    This is the most general constructor; it can be used to define any celestial WCS
+    as long as it contains the @ref skywcs_frames "required frames".
+
+    @throw lsst::pex::exceptions::InvalidParameterError if `frameSet` is missing
+    any of the @ref skywcs_frames "required frames".
+    */
+    explicit SkyWcs(ast::FrameSet const & frameSet);
+
+    ~SkyWcs(){};
+
+    /**
+    Get the pixel scale at the specified pixel position
+
+    The scale is the square root of the area of the specified pixel on the sky.
+    */
+    Angle getPixelScale(Point2D const &pixel) const;
+
+    /**
+    Get CRPIX, the pixel origin, using the LSST convention
+    */
+    Point2D getPixelOrigin() const;
+
+    /**
+    Get CRVAL, the sky origin or celestial fiducial point
+    */
+    SpherePoint getSkyOrigin() const;
+
+    /**
+    Get the CD matrix at the pixel origin
+    */
+    Eigen::Matrix2d getCdMatrix() const;
+
+    /**
+    Get a local TAN WCS approximation to this WCS at the specified pixel position
+    */
+    SkyWcs getTanWcs(Point2D const &pixel) const;
+
+    /**
+    Return a copy of this SkyWcs with the pixel origin by the specified amount
+    */
+    SkyWcs shiftedPixelOrigin(double dx, double dy) const;
+
+    /**
+    Compute sky position(s) from pixel position(s)
+
+    This is another name for tranForward, plus an overload that takes a pair of doubles.
```

Comment:
I agree that the names are less than ideal, and we are going to have to have this discussion. But I'd rather keep it out of this ticket.

Transform is already in the stack and already has tranForward and tranInverse. This is a subclass of that. I don't think I can responsibly hide those methods, whatever we end up calling them.